app/controllers/admin/vouchers_controller.rb

```diff
@@ -9,19 +9,33 @@ def new
     end
 
     def create
-      @voucher = Voucher.create(permitted_resource_params.merge(enterprise: @enterprise))
+      case params[:voucher][:voucher_type]
+      when "Vouchers::FlatRate"
+        @voucher =
+          Vouchers::FlatRate.create(permitted_resource_params.merge(enterprise: @enterprise))
+      when "Vouchers::PercentageRate"
+        @voucher =
+          Vouchers::PercentageRate.create(permitted_resource_params.merge(enterprise: @enterprise))
+      else
+        @voucher.errors.add(:type)
+        return render_error
+      end
 
       if @voucher.save
-        flash[:success] = flash_message_for(@voucher, :successfully_created)
+        flash[:success] = Spree.t(:successfully_created, resource: "Voucher")
```

Comment:
I would love to phase out the use of "Spree". This is not a Spree controller.

Reply:
Done, I saw that this translation also exists at the root level